content/ADR-258-trigger-areas.md

```diff
@@ -0,0 +1,193 @@
+- `CL_CUSTOM2`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM3`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM4`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM5`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM6`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM7`: A custom layer for any other object that can move around the scene
+- `CL_CUSTOM8`: A custom layer for any other object that can move around the scene
+- `CL_PLAYER`: The default layer for the player
+
+A single trigger area can have multiple trigger layers at once, similarly to how collision layers work on colliders.
+
+When using the `POINTER` layer, the trigger area will trigger events when the pointer is hovered over it, in the same way that ON_HOVER events from the PointerEvents component work.
+
+When using the `PHYSICS` layer, the trigger area will trigger events when any entity with a physics collider overlaps with it.
+
+Note: If a collider with the same shape blocks a player from entering a trigger area, then we won't consider a trigger event being sent. There must be an overlap with the area, not just contact. For example, if a same entity has both a MeshCollider component set to the layer `CL_PHYSICS` and a Trigger component set to the layer `CL_PLAYER`, and both meshes have the same shape, then the player will never be able to activate a trigger event from this entity. In the future we'll likely want to do _Collision Events_ as a separate feature.
+
+### Trigger events
+
+Trigger areas can trigger events when the player (or any other entity on the trigger layer) enters, exits or stays in the area.
+
+Trigger events would have to be shared from the engine to the SDK via a component, following a similar approach as we do with pointer events and raycasts. We should create a `TriggerCollisionResult` component. Creators are not expected to read values or make use of this component in any way, unless they really want to fine tune their scene’s behavior.
+
+This component will be added by the engine, similarly to how pointer events are handled. The event will be triggered when the entity enters, exits or stays in the area. Each event will have the following fields:
+
+- `triggeredEntity`: The entity that was triggered (this is the entity that owns the trigger area)
+- `triggeredEntityPosition`: The position of the triggered entity at the time of the trigger.
+- `triggeredEntityRotation`: The rotation of the triggered entity at the time of the trigger.
+- `state`: The state of the trigger event (ENTER, EXIT, STAY)
+- `timestamp`: The timestamp of the trigger event
+- `trigger`: An object with the following fields:
+  - `entity`: The entity that triggered the trigger
+  - `layer`: The collision layer of the entity that triggered the trigger (this is the same as the trigger layer of the trigger area)
```

Comment:
Suggesting based on what we ended up implementing for [TriggerAreaResult component](https://github.com/decentraland/protocol/blob/258fabc1584701e63f129af3ca9895960305f703/proto/decentraland/sdk/components/trigger_area_result.proto):
```suggestion
  - `layers`: The collision layermask of the entity that triggered the Trigger Area
```